### ISSUE TITLE
Serialization of type Map raise IndexOutOfBoundsException

### ISSUE THREAD

**Issue by ptgreg** [json4s/json4s#276]
Hi,

I have `java.lang.IndexOutOfBoundsException: 1` when I run this sample code:

```
type Counters = Map[String, Double]
case class A(counters: Counters)
val a = A(Map("cnt" -> 1.0))
import org.json4s.native.Serialization
implicit val formats = org.json4s.DefaultFormats
Serialization.write(a)
```

If I remove the `type Counters...`

```
case class A(counters: Map[String, Double])
val a = A(Map("cnt" -> 1.0))
import org.json4s.native.Serialization
implicit val formats = org.json4s.DefaultFormats
Serialization.write(a)
```

It works.

Any idea?

**Comment by seratch** [comment]
This looks a similar problem like #281

**Comment by yohanakh** [comment]
Hey,

Any news on this?
I have the same issue where Counters is:
case class Counters extends HashMap[String, String].
Exception in thread "main" org.json4s.package$MappingException: unknown error
Caused by: java.lang.IndexOutOfBoundsException: 0

I see that if I declare Counters as:
case class Counters[A,B] extends HashMap[String,String]
then it works fine.
This is the problematic line:
https://github.com/json4s/json4s/blob/3.6/core/src/main/scala/org/json4s/Extraction.scala#L370
Counters in the first example isMap but has no typeArgs.

Thanks

**Comment by mccartney** [comment]
I assume this has been fixed in #577.
